com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/PollExecuteAppStatusExecutionTest.java

```diff
@@ -0,0 +1,178 @@
+                                     null, null, true, AsyncExecutionState.ERROR),
+                        // (7) Application with Custom failure marker
+                        Arguments.of(new ApplicationLog(APPLICATION_GUID, "FAILURE", new Date(START_TIME_IN_MILLIS), MessageType.STDERR, APP_SOURCE, SOURCE_ID),
+                                     null, "execution failure", false, AsyncExecutionState.RUNNING),
+                        // (8) Log message with non APP Source
+                        Arguments.of(new ApplicationLog(APPLICATION_GUID, "info service", new Date(START_TIME_IN_MILLIS), MessageType.STDOUT, "service", SOURCE_ID),
+                                     null, null, false, AsyncExecutionState.RUNNING)
+            );
+        //@formatter:on
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    public void testStep(ApplicationLog applicationLog, String successMarker, String failureMarker, boolean shouldStopApp,
+                         AsyncExecutionState expectedExecutionState) {
+        CloudApplicationExtended application = buildApplication(successMarker, failureMarker, shouldStopApp);
+        prepareContext(application);
+        prepareRecentLogsRetriever(applicationLog);
+        prepareStepLogger();
+        prepareClientProvider();
+
+        AsyncExecutionState resultState = step.execute(executionWrapper);
+
+        assertEquals(expectedExecutionState, resultState);
+
+        if (shouldStopApp) {
+            verify(client).stopApplication(application.getName());
+        } else {
+            verify(client, never()).stopApplication(application.getName());
+        }
+    }
+
+    private CloudApplicationExtended buildApplication(String successMarker, String failureMarker, boolean shouldStopApp) {
+        Map<String, Object> deployAttributes = new HashMap<>();
+        deployAttributes.put(SupportedParameters.STOP_APP, shouldStopApp);
+        if (successMarker != null) {
+            deployAttributes.put(SupportedParameters.SUCCESS_MARKER, successMarker);
+        }
+        if (failureMarker != null) {
+            deployAttributes.put(SupportedParameters.FAILURE_MARKER, failureMarker);
+        }
+
+        Map<String, String> applicationEnv = MapUtil.asMap(Constants.ENV_DEPLOY_ATTRIBUTES, JsonUtil.toJson(deployAttributes));
+        return ImmutableCloudApplicationExtended.builder()
+                                                .name("test-app")
+                                                .env(applicationEnv)
+                                                .build();
+    }
+
+    private void prepareContext(CloudApplicationExtended application) {
+        StepsUtil.setApp(context, application);
+        StepsUtil.setAppStateActionsToExecute(context, new HashSet<>(Arrays.asList(ApplicationStateAction.EXECUTE)));
+        context.setVariable(com.sap.cloud.lm.sl.cf.process.Constants.VAR_USER, USER_NAME);
+        long startTime = TimeUnit.SECONDS.toMillis(LocalDateTime.of(2019, 1, 1, 0, 0)
```

Comment:
A bit better than stating the seconds explicitly but it will be better still if the arguments are specified like
Calendar(2019, AUGUST, 1, ...)
LocalDateTime.of()'s arguments aren't explicitly clear which is which (if the first '1' is the month or the day) without consulting the docs